src/thread/SDL_thread_c.h

```diff
@@ -38,8 +38,6 @@
 #include "vita/SDL_systhread_c.h"
 #elif defined(SDL_THREAD_N3DS)
 #include "n3ds/SDL_systhread_c.h"
-#elif defined(SDL_THREAD_STDCPP)
```

Comment:
We may want to keep the std:: thread implementation. I can easily see people wanting to use that in SDL in the future.

Reply:
The only thing that needed it was WinRT, and for any other platform, it has limitations that the underlying platform APIs do not. I'm deeply in favor of deleting it. We can restore it if something shows up that needs it in the future.

Reply:
Okay, sounds good.